[DOC-3069] Mark as failure (#1716)
harness/developer-hub: docs/platform/8_Pipelines/mark-as-failed.md

---
title: Mark stage as failed
description: Mark a stage as failed during execution.
sidebar_position: 9
---

> You can mark a specific stage as failed during pipeline execution by selecting the `Mark Stage As Failed` option.

Reviewer: The label is "Mark as Failure", I believe. @sahilhindwani-harness ?

Reviewer: Apologies - no, it's "Mark Stage as Failed". My miss.,

This lets you do the following: 
- Abort a step that is running.

> - Trigger rollback after the step is aborted.

Reviewer: Basically we can trigger whatever failure strategy is configured, including Rollback or Retry or anything else.

## Important

- You must have `Execute` pipeline permission to be able to mark as stage as failed.
- The option `Mark Stage As Failed` is at a stage level in the execution phase. During the execution of parallel stages (like a matrix), you must either mark each one as a failure or let the other steps complete before applying the failure strategy.
- Harness aborts any delegate task currently running after you mark a stage as failed.

## Mark a stage as failed

To mark a stage as failed during pipeline execution: 

> 1. Select your pipeline and then select **Run**.

Reviewer 2: ```suggestion
1. Select your pipeline, and then select **Run**.
```

> 2. Enter runtime input if any and select **Run Pipeline**.

Reviewer 2: ```suggestion
2. Enter runtime input, if any, and then select **Run Pipeline**.
```

3. Select **Mark Stage As Failed**.
   
   ![](./static/mark-as-failed-option.png)

4. Select **Confirm**.

>    The **Error Summary** for the stage displays `User Initiated Failure`. The failure type of this stage is set to `USER_MARKED_FAILURE`.

Reviewer 2: ```suggestion
   
   The **Error Summary** for the stage displays `User Initiated Failure`. The failure type of this stage is set to `USER_MARKED_FAILURE`.
```

   ![](./static/error-summary.png)

## Configure failure strategy

To configure failure strategy for a stage marked as failure: 

> 1. Select your pipeline and then select the stage for which you want to configure failure strategy.

Reviewer 2: ```suggestion
1. Select your pipeline, and then select the stage for which you want to configure failure strategy.
```

2. Select **Advanced**.
3. In **Failure Strategy**, select **Add**.
4. Select **User Marked Failure** in **On failure of type**.

> 5. Select desired action in **Perform Action** like **Rollback**, **Retry**.

Reviewer 2: ```suggestion
5. Select desired action in **Perform Action**, for example, **Rollback** or **Retry**.
```

>    The action you select here gets triggered after you mark a stage as failed.

Reviewer: "You can select Rollback, Retry, or other options to be executed at this point."

Author: @SudarshanPurohit **Perform Action** has all the options like Rollback, Retry, etc. Do we need to mention the options again specifically?

Reviewer: Yes, I see the previous line. Let's keep it as it is.

6. Select **Save**.

   ![](./static/failure%20strategy.png)

:::important

> If you mark a stage as failed whose failure strategy is not defined, Harness triggers the default failure strategy.

Reviewer 2: ```suggestion
If you mark a stage as failed, and its failure strategy is not defined, Harness triggers the default failure strategy.
```
